Afterpay countries

LeasoftKonstantinShapoval opened the pull request.

shapovalkv committed:
Added hiding payment method tools for AfterPay with available countries in admin panel.

shapovalkv committed:
Added hiding payment method tools for AfterPay with available countries in admin panel.

shapovalkv committed:
Change version of plugin for 1.5.1

LeasoftKonstantinShapoval committed:
Added Nl, Fr, De languages for AfterPay countries.
Added validation for isset AfterPay values in country validation.

LeasoftKonstantinShapoval committed:
Deleted const, function and validation in twig file.
Added validation for default values in catalog\controller for country validation.

LeasoftKonstantinShapoval committed:
Changed the text in localization files for AfterPay.
Added values for "Countries available for AfterPay" in event of installation AfterPay payment.
Updated CHANGELOG.md and README.md.

LeasoftKonstantinShapoval committed:
Change info_default_country_access to info_example_country_access in all code.

LeasoftKonstantinShapoval committed:
Delete one useless if in catalog\model\extension\payment\emspay_afterpay.php
Made code Refactoring.

tiutiunnykleasoft commented on upload/admin/language/de-de/extension/payment/emspay_ideal.php:
```diff
@@ -52,4 +52,4 @@
 $_['entry_klarna_test_api_key'] = 'Test API schlüssel:';
-$_['info_help_klarna_test_api_key'] = 'Wenn eingegeben, wird dieser API-Schlüssel nur zu Testzwecken verwendet werden.';
\ No newline at end of file
+$_['info_help_klarna_test_api_key'] = 'Wenn eingegeben, wird dieser API-Schlüssel nur zu Testzwecken verwendet werden.';
```
Remove empty string, that file isn't changed

tiutiunnykleasoft commented on upload/admin/language/fr-fr/extension/payment/emspay_ideal.php:
```diff
@@ -52,4 +52,4 @@
 $_['entry_klarna_test_api_key'] = 'clé API de Test:';
-$_['info_help_klarna_test_api_key'] = 'Si entré, cette clé API sera utilisée uniquement à des test.';
\ No newline at end of file
+$_['info_help_klarna_test_api_key'] = 'Si entré, cette clé API sera utilisée uniquement à des test.';
```
Remove empty string

tiutiunnykleasoft commented on upload/admin/language/nl-nl/extension/payment/emspay_ideal.php:
```diff
@@ -52,4 +52,4 @@
 $_['entry_klarna_test_api_key'] = 'Test API Key:';
-$_['info_help_klarna_test_api_key'] = 'Vul hier de API Key in van uw test webshop.';
\ No newline at end of file
+$_['info_help_klarna_test_api_key'] = 'Vul hier de API Key in van uw test webshop.';
```
remove empty string

tiutiunnykleasoft commented on upload/catalog/model/extension/payment/emspay_afterpay.php:
```diff
@@ -2,17 +2,17 @@
+            WHERE geo_zone_id = '" . (int)$this->config->get('emspay_afterpay_geo_zone_id') . "' 
+            AND country_id = '" . (int)$address['country_id'] . "' 
+            AND (zone_id = '" . (int)$address['zone_id'] . "' 
             OR zone_id = '0');"
```
Do these changes actually change something?
If not, remove them

LeasoftKonstantinShapoval replied:
Roll backed.

tiutiunnykleasoft commented on upload/system/library/emshelper.php:
```diff
@@ -658,6 +658,22 @@ public static function ipIsEnabled($ipList)
+     */
+    public static function CountryValidator($countryList, $billingAddress)
+    {
+        if ($countryList !== '' ) {
```
https://github.com/emspay/ems-online-opencart-2/pull/7#discussion_r465632188

tiutiunnykleasoft commented on upload/system/library/emshelper.php:
```diff
@@ -658,6 +658,22 @@ public static function ipIsEnabled($ipList)
+     * @param $countryList
+     * @return bool
+     */
+    public static function CountryValidator($countryList, $billingAddress)
```
https://github.com/emspay/ems-online-opencart-2/pull/7#discussion_r465632814

tiutiunnykleasoft commented on upload/admin/view/template/extension/payment/emspay.twig:
```diff
@@ -284,6 +284,24 @@
+                            </div>
+                        </div>
+                    {% endif %}
+
```
https://github.com/emspay/ems-online-opencart-2/pull/7#discussion_r465779885

LeasoftKonstantinShapoval committed:
Refactored code:
- Added logic for emty "Countries available for AfterPay" field.
- Transferred AfterPay countries availability field in ems.twig.
- Removed all empty string.
- Changed localization for AftePay countries availability.

LeasoftKonstantinShapoval committed:
Remove empty sting in localization files.

LeasoftKonstantinShapoval committed:
Remove empty sting in localization files.
Rollback changes in upload/catalog/model/extension/payment/emspay_afterpay.php

LeasoftKonstantinShapoval committed:
Refactored return countryValidator

LeasoftKonstantinShapoval committed:
Deleted needless return.
